src/client/BuilderClient.ts

```diff
@@ -89,12 +89,12 @@ export class BuilderClient {
   }
 
   /**
-   * Updates or inserts an item.
+   * Updates or inserts an item. The item can be updated by id or URN.
    * @param item - The item to insert or update.
    * @param newContent - The content to be added or updated in the item. This content must be contained in the items contents.
    */
   public async upsertItem(
-    item: LocalItem,
+    item: Omit<LocalItem, 'id'> & { id?: LocalItem['id'] },
```

Comment:
Would you mind adding a test for this new behavior?

Reply:
yup, sure! tests added!